python/src/scipp/neutron/instrument_view.py

```diff
@@ -0,0 +1,464 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2019 Scipp contributors (https://github.com/scipp)
+# @author Neil Vaytet
+
+# Scipp imports
+from ..config import plot as config
+from ..plot.render import render_plot
+from ..plot.sciplot import SciPlot
+from ..plot.sparse import histogram_sparse_data, make_bins
+from ..plot.tools import parse_params
+from ..utils import name_with_unit, value_to_string
+from .._scipp import core as sc
+
+# Other imports
+import numpy as np
+import ipywidgets as widgets
+import matplotlib.pyplot as plt
+from matplotlib import cm
+try:
+    import ipyvolume as ipv
+except ImportError:
+    ipv = None
+import IPython.display as disp
+from matplotlib.collections import PatchCollection
+from matplotlib.patches import Rectangle
+
+
+def instrument_view(scipp_obj=None, bins=None, masks=None, filename=None,
+                    figsize=None, aspect="equal", cmap=None, log=False,
+                    vmin=None, vmax=None, size=0.1, projection="3D",
+                    nan_color="#d3d3d3", continuous_update=True,
+                    dim=sc.Dim.Tof):
+    """
+    Plot a 2D or 3D view of the instrument.
+    A slider is also generated to navigate the time-of-flight dimension.
```

Comment:
Does it work if it is not `Dim.Tof` but something like `Dim.Wavelength`? Apparently, since you have the `dim` arg, just the docstring is misleading?

Reply:
yes i made this change right at the end and then forgot to change the docstring